components/ai_chat/common/mojom/ai_chat.mojom

```diff
@@ -63,6 +63,7 @@ interface PageHandler {
   ClearConversationHistory();
   RetryAPIRequest();
   GetAPIResponseError() => (APIError error);
+  UserHasValidPremiumCredential() => (bool result);
```

Comment:
nit: `GetHasValidPremiumCredential()` (or just `GetIsPremium()` unless we're also planning to add something similar to distinguis "is premium" vs "is premium with valid credential"